components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -1646,6 +1657,9 @@ interface BraveWalletService {
 
   // Resets the keyring and the related preferences
   Reset();
+
+  // Used for UI to trigger the allowance discovery
+  DiscoverEthAllowances();
```

Comment:
Since our plan is make this triggered by an explicit UI, and won't be something running in the background, it seems better to me to change this to `DiscoverEthAllowances() => (array<AllowanceInfo> allowances)` to have result directly in the callback and not via event listener.
In addition to allowances, I think it might be useful to have an error and error_message returned in the callback too. For example, UI can know the call wasn't successful due to in-progress operation, or something else.
Wdyt?